scripts/create-docs.py

```diff
@@ -0,0 +1,42 @@
+import os
+import re
+import sys
+
+lib_dir = "lib"
+output_resource = sys.argv[1] if len(sys.argv) > 1 else "all-keywords.robot"
+
+# Collect all .robot files
+robot_files = ["keywords.robot"]
+for root, dirs, files in os.walk(lib_dir):
+    for file in files:
+        if file.endswith(".robot"):
+            robot_files.append(os.path.join(root, file))
+
+# Regex pattern to detect keyword definitions
+keyword_pattern = re.compile(
+    r"\n(?:(?!Library|Resource|Variables|Documentation)[A-Z][a-z0-9_]+[\ ]+.*)"
+)
+
+delimiter = "."  # Between filename and the keyword
+
+# Create the combined resource file with prefixed keywords
+with open(output_resource, "w", encoding="utf8") as res_file:
+
+    for robot_file in robot_files:
+        file_prefix = os.path.splitext(os.path.basename(robot_file))[0] + delimiter
+        index_offset = 1  # Initial value = length of \n
+
+        with open(robot_file, "r", encoding="utf8") as file:
+            text = file.read()
+
+            for m in re.finditer(keyword_pattern, text):
+                start = m.start(0)
+                text = (
+                    text[: start + index_offset]
+                    + file_prefix
+                    + text[start + index_offset :]
+                )
+                index_offset += len(file_prefix)
```

Comment:
Not sure if modifying text you are looping through is such a good idea if there are easier ways.
Maybe you could read each file line by line and check if line matches pattern? Working example

```py
import os
import re
import sys

lib_dir = "lib"
output_resource = sys.argv[1] if len(sys.argv) > 1 else "all-keywords.robot"

# Collect all .robot files
robot_files = ["keywords.robot"]
for root, dirs, files in os.walk(lib_dir):
    for file in files:
        if file.endswith(".robot"):
            robot_files.append(os.path.join(root, file))

# Regex pattern to detect keyword definitions
keyword_pattern = re.compile(
    r"^(?:(?!Library|Resource|Variables|Documentation)[A-Z][a-z0-9_]+[\ ]+.*)",
    re.MULTILINE
)

delimiter = "."  # Between filename and the keyword

with open(output_resource, "w", encoding="utf8") as write_file:
    for robot_file in robot_files:
        file_prefix = os.path.splitext(os.path.basename(robot_file))[0] + delimiter
        index_offset = 0

        with open(robot_file, "r", encoding="utf8") as read_file:
            for line in read_file:
                if re.match(keyword_pattern, line):
                    write_file.write(file_prefix + line)
                else:
                    write_file.write(line)
```

Reply:
I think it's a little easier to modify that way (unless we need multi-line processing)

Reply:
If you implement reading line by line then you don't likely don't need `re.MULTILINE` flag in keyword_patter

Reply:
Fixed in 5830fdb4f34817ec39ebdf1d340b5d3f2f1de161

Reply:
Now redundant multiline argument also was removed in 515779bb9ff69645ff471f72796e0dca3b3062da